BOSA21Q1-492 Fix tests for user sign in, initiative creation and voting

sergei-krylov opened the pull request.

simon-lejeune commented:
quick remarks, sorry if missed some things:

- last time i checked, the test suite didn't run on master but only in 0.24 branch, did that change?
- i see you're making changes in create_initiative_spec.rb and update_initiative_spec.rb, are they originally copy pasted from decidim? did you run a diff on both files now? did we diverge much?
- one of the point in the task is "Test a user can connect with email and password", i don't see that here

don't you think it would be better to realign the tests copy pasted from decidim with decidim and add a new "test flow" creating a user from scratch, login with is password and then create a petition and vote on it with that new user? it would be helpful to start writing some flow tests now, anyway we'll need to do that for all the connexions flow

simon-lejeune commented on spec/decidim-initiatives/spec/commands/decidim/initiatives/vote_initiative_spec.rb:
```diff
@@ -186,63 +265,6 @@ module Initiatives
           end
 
-          # (?) turn off this part as there is no `document_number` any more (see VoteFormExtend `metadata` method)
-          xcontext "when initiative type has document number authorization handler" do
```
did we change that in our extensions to decidim?

sergei-krylov replied:
No, we just don't use this auth handler with initiatives. The test has been commented out previously (and was failing), so I just removed it as for now.

sergei-krylov commented on spec/decidim-initiatives/spec/commands/decidim/initiatives/vote_initiative_spec.rb:
```diff
@@ -107,10 +107,18 @@ module Initiatives
 
         context "when support threshold is reached" do
-          let(:admin) { create(:user, :admin, :confirmed, organization: organization) }
+          let!(:admin) { create(:user, :admin, :confirmed, organization: organization) }
```
We need that to create an admin user right away, otherwise `let` is doing that in lazy way, and it breaks the test when it checks for the organization admins in some other place (not directly in test).

sergei-krylov commented:
@simon-lejeune , yes the tests are already fixed in upgraded 0.24 and we have a green bar there. I just fixed them here in current 0.22 version as well, so in case we made some changes we can double check these specific tests (at least).
Yes they're originally copy-pasted from original decidim and updated to reflect changes made in extensions. There are some differences in them of course.
The test that user can connect with email\password is in authorizations_spec.rb and it is wokring, additionally I fixed the checks for handler name to make the whole file pass.
Regarding realigning the tests - I don't think it's worth it to start on older version, and overall - we still didn't find the proper way of how to run the original tests with some changes/extends on top of them to be able to reuse original tests as much as possible... I think we can leave it to that part.

simon-lejeune reviewed (approved):
Thank you for the clarifications. I'm still thinking of a way to get back the OG decidim test cases somehow..
